src/colorizer/CanvasWithOverlay.ts

```diff
@@ -109,325 +107,142 @@ export default class CanvasWithOverlay extends ColorizeCanvas {
   }
 
   public setSize(width: number, height: number): void {
-    this.canvasWidth = width;
-    this.canvasHeight = height;
+    this.canvasSize.x = width;
+    this.canvasSize.y = height;
     super.setSize(width, height);
   }
 
-  // Rendering ////////////////////////////////
-
-  private getPixelRatio(): number {
-    return window.devicePixelRatio || 1;
+  public getIdAtPixel(x: number, y: number): number {
+    const headerHeight = this.headerSize.y;
+    return super.getIdAtPixel(x, y - headerHeight);
   }
 
-  updateScaleBarOptions(options: Partial<ScaleBarOptions>): void {
-    this.scaleBarOptions = { ...this.scaleBarOptions, ...options };
-  }
+  // Getters/Setters ////////////////////////////////
 
-  updateTimestampOptions(options: Partial<TimestampOptions>): void {
-    this.timestampOptions = { ...this.timestampOptions, ...options };
+  updateScaleBarStyle(style: Partial<ScaleBarStyle>): void {
+    this.scaleBarStyle = { ...this.scaleBarStyle, ...style };
   }
 
-  updateBackgroundOptions(options: Partial<OverlayFillOptions>): void {
-    this.backgroundOptions = { ...this.backgroundOptions, ...options };
+  updateTimestampStyle(style: Partial<TimestampStyle>): void {
+    this.timestampStyle = { ...this.timestampStyle, ...style };
   }
 
-  private getTextDimensions(ctx: CanvasRenderingContext2D, text: string, options: StyleOptions): Vector2 {
-    ctx.font = `${options.fontSizePx}px ${options.fontFamily}`;
-    ctx.fillStyle = options.fontColor;
-    const textWidth = ctx.measureText(text).width;
-    return new Vector2(textWidth, options.fontSizePx);
+  updateInsetBoxStyle(style: Partial<InsetBoxStyle>): void {
+    this.insetBoxStyle = { ...this.insetBoxStyle, ...style };
   }
 
-  /**
-   * Renders text on the canvas, using its bottom right corner as the origin.
-   * @param ctx the canvas context to render to.
-   * @param text the text to render.
-   * @param originPx the origin of the text, from the lower right corner, in pixels.
-   * @param options style options for the text.
-   * @returns the width and height of the text, as a Vector2.
-   */
-  private renderRightAlignedText(
-    ctx: CanvasRenderingContext2D,
-    originPx: Vector2,
-    text: string,
-    options: StyleOptions
-  ): Vector2 {
-    ctx.font = `${options.fontStyle} ${options.fontSizePx}px ${options.fontFamily}`;
-    ctx.fillStyle = options.fontColor;
-    const textWidth = ctx.measureText(text).width;
-    // Magic number to nudge text up a bit so it looks vertically centered.
-    const textOffset = Math.round(options.fontSizePx * 0.1);
-    ctx.fillText(text, this.canvas.width - textWidth - originPx.x, this.canvas.height - originPx.y - textOffset);
-    return new Vector2(textWidth, options.fontSizePx);
```

Comment:
Replaced with `configureCanvasText` and `renderCanvasText` in `utils.ts`, which is a lot more flexible!